graph-kiota/graph-kiota.md

```diff
@@ -0,0 +1,126 @@
+
+This library contains the
+- Factory method/classes returning the:
+    - Core client with the user configurations such as authprovider, custom middleware, request options.
+    - HttpCore instance for the service library with user configurations such as authprovider, custom middleware, request options.
+
+- Tasks
+    - PageIteratorTask
+    - LargeFileUploadTask
+    - Batching
+
+- Authentication Providers
+    - Use the `IAuthProvider` and  auth provider class from the Kiota Auth abstractions.
+
+
+#### Graph Service library specifications
+
+- Discuss about ways to add custom code or custom configurations to the generated `GraphServiceClient` code.
+- The custom configurations that can go here are as follows: 
+ - Setting the default for `HttpCore` with a Graph specific `HttpCore` instance. 
+ - Adding multiple constructors to the `GraphServiceClient`. 
+
+#### Graph Middlewares
+
+- Import middlewares from Kiota core library
+- How to add graph specific checks in the middleware. Example: The JS SDK checks if the url is a Graph url or a custom host passed by the user before appending telemetry headers or auth headers. If not, the headers are removed.
+
+- AuthHandler - Kiota core
+- RetryHandler - Kiota core
+- RedirectHandler - Kiota core
+- CompressionHandler - Kiota core
+- CachingHandler - Kiota core
+- ChaosHandler - Graph core 
+#### Use cases
+
+- Graph core client 
+```
+var response = new coreClient.requestURL("/me").get();
+```
+
+- Graph service client
+```
+var response = new serviceClient.me().get();
+```
+
+- Graph service client calling a core functionality
+
+- The service library should extend the capabilities of the core library, that is, a user using service client should not require to create or use a core client to access any functionality. 
+```
+var response = new serviceClient.requestURL("/me").get(); // Example : a user of the JS SDK should be able to access GraphServiceClient.api(/me) without initialising the core.
+```
+
+- Graph core client with a task
+
+```
+var response = new coreClient.requestURL("/me").get();
+
+PageIterator(coreclient, response)
+```
+
+- Graph service client with a task
+
+```
+var response = new serviceClient.me().get();
+PageIterator(serviceClient, response)
+```
+
+- The service library should extend the capabilities of the core library, that is, a user using service client should not require to create or use a core client to access any functionality.
+-  Some options to  achieve this (TBD):
+ -  Maintain a parent abstract class for the core and the service client.
+ -  `GraphServiceClient` extends the `GraphCoreClient`.
+
+### More questions
+
+- Does the generated ApiClient have to be tightly coupled with the generated request builders? Should it always be tightly coupled? [MEM] I ask as there could be scenarios where the request builders are not necessary and the client developers just wants the Kiota.Core functionality in the client, and will use client.requestUrl<Response>.Get();
```

Comment:
the api client is only a request builder with a special constructor and fewer properties. So in that sense yes. This was an early design choice.